pkg/kubelet/stats/cadvisor_stats_provider.go

```diff
@@ -270,26 +270,34 @@ func (p *cadvisorStatsProvider) ImageFsStats(ctx context.Context) (imageFsRet *s
 		return imageFs, imageFs, nil
 	}
 	imageStats, err := p.imageService.ImageFsInfo(ctx)
-	if err != nil || imageStats == nil {
-		return nil, nil, fmt.Errorf("failed to get image stats: %v", err)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to get image stats: %w", err)
+	}
+	if imageStats == nil || len(imageStats.ImageFilesystems) == 0 || len(imageStats.ContainerFilesystems) == 0 {
+		return nil, nil, fmt.Errorf("incomplete image stats: %+v", imageStats)
```

Comment:
To make it more explicit, perhaps.

```suggestion
		return nil, nil, fmt.Errorf("missing image stats: %+v", imageStats)
```

Reply:
Fixed as suggested.